lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -586,6 +586,21 @@ static bool isExpressionUnableToInline(Operation *op) {
   if (isa<StructCreateOp>(op))
     return true;
 
+  // If the aggregate constant contains a struct then we must spill it.
+  if (auto aggConstantOp = dyn_cast<AggregateConstantOp>(op)) {
+    Type type = aggConstantOp.getType();
+    while (true) {
+      if (auto arrayType = type.dyn_cast<ArrayType>())
+        type = arrayType.getElementType();
+      else if (auto uarrayType = type.dyn_cast<UnpackedArrayType>())
+        type = uarrayType.getElementType();
+      else if (auto structType = type.dyn_cast<StructType>())
+        return true;
+      else
+        return false;
```

Comment:
`break` would be correct here.

Reply:
I have deleted this code block in the newest version and now always spill for this operation.